openpgp/packet/signature.go

```diff
@@ -676,6 +762,66 @@ func (sig *Signature) signPrepareHash(h hash.Hash) (digest []byte, err error) {
 	return
 }
 
+// PrepareSign must be called to create a hash object before Sign for v6 signatures.
+// The created hash object initially hashes a randomly generated salt
+// as required by v6 signatures. The generated salt is stored in sig. If the signature is not v6, 
+// the method returns an empty hash object.
+// See RFC the crypto refresh Section 3.2.4.
+func (sig *Signature) PrepareSign(config *Config) (hash.Hash, error) {
+	if !sig.Hash.Available() {
+		return nil, errors.UnsupportedError("hash function")
+	}
+	hasher := sig.Hash.New()
+	if sig.Version == 6 {
+		if sig.salt == nil {
+			var err error
+			sig.salt, err = SignatureSaltForHash(sig.Hash, config.Random())
+			if err != nil {
+				return nil, err
+			}
+		}
+		hasher.Write(sig.salt)
+	}
+	return hasher, nil
+}
+
+// PrepareSignWithSalt stores a signature salt within sig for v6 signatrues.
+// Assumes salt is generated correctly and checks if length matches.
+// If the signature is not v6, the method ignores the salt.
+// Use PrepareSign whenever possible instead of this method.
+// See RFC the crypto refresh Section 3.2.4.
+func (sig *Signature) PrepareSignWithSalt(salt []byte) (error) {
+	if sig.Version == 6 {
+		expectedSaltLength, err := SaltLengthForHash(sig.Hash)
+		if err != nil {
+			return err
+		}
+		if salt == nil || len(salt) != expectedSaltLength {
+			return errors.InvalidArgumentError("unexpected salt size for the given hash algorithm")
+		}
+		sig.salt = salt
+	}
+	return nil
+}
```

Comment:
This function is a bit confusing to me as it's similarly named to the previous one but does something very different. So I would either name it differently, or just inline this into the place where it's used, as it seems a bit specific.